### PR TITLE
Refactors data to datum

### PR DESCRIPTION
**Pull request by cpb**
Depends on FormidableLabs/victory-core#26
Closes FormidableLabs/victory-chart#116

### CODE REVIEW

File: src/components/victory-area/area-label.jsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { VictoryLabel, Helpers } from "victory-core";
 
 export default class AreaLabel extends React.Component {
   static propTypes = {
-    data: PropTypes.array,
```
Comment (boygirl) [inline src/components/victory-area/area-label.jsx]: Actually area gets the whole dataset, so data is appropriate here.

File: src/components/victory-area/area.jsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { Helpers } from "victory-core";
 
 export default class Area extends React.Component {
   static propTypes = {
-    data: PropTypes.array,
```
Comment (boygirl) [inline src/components/victory-area/area.jsx]: ditto

File: src/components/victory-line/line-label.jsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { VictoryLabel, Helpers} from "victory-core";
 
 export default class LineLabel extends React.Component {
   static propTypes = {
-    data: PropTypes.array,
```
Comment (boygirl) [inline src/components/victory-line/line-label.jsx]: Line gets the whole data set

File: src/components/victory-axis/victory-axis.jsx [hunk 4]
```diff
@@ -324,6 +324,7 @@ export default class VictoryAxis extends React.Component {
       y: componentProps.y || y,
       textAnchor: componentProps.textAnchor || "middle",
       verticalAnchor: componentProps.verticalAnchor || verticalAnchor,
+      datum: props.datum,
```
Comment (boygirl) [inline src/components/victory-axis/victory-axis.jsx]: The axis label doesn't get data

**Comment by cpb** [comment]
- [ ] AreaLabel actually receives the whole dataset.
- [ ] Area likewise receives the whole dataset.
- [ ] LineLabel receives the whole dataset.
- [ ] VictoryAxis doesn't receive data.

**Comment by boygirl** [comment]
cc/ @cpb since I made merge conflict hell for you, I decided to just scoop up these changes in another branch I was working on. I was touching all the same code as a last once over on my events stuff. Sorry!

**Pull request closed without merging**